Shared/Samples/Show viewshed from point on map/ShowViewshedFromPointOnMapView.swift

```diff
@@ -0,0 +1,202 @@
+            // Clears previously viewshed drawing.
+            resultGraphicsOverlay.removeAllGraphics()
+            // If there is a geoprocessing job in progress it is cancelled.
+            await geoprocessingJob?.cancel()
+            guard let spatialReference = point.spatialReference else { return }
+            // Creates a feature collection table based on the spatial reference for the tapped location
+            // on the map.
+            let featureCollectionTable = FeatureCollectionTable(
+                fields: [Field](),
+                geometryType: Point.self,
+                spatialReference: spatialReference
+            )
+            do {
+                // Creates a feature for the point tapped.
+                let feature = featureCollectionTable.makeFeature(geometry: point)
+                // Asynchronously adds that feature to the table.
+                try await featureCollectionTable.add(feature)
+                try await performGeoprocessing(featureCollectionTable)
+            } catch {
+                throw error
+            }
+        }
+        
+        /// Contains the logic for the geoprocessing and passes the result on to another function to display.
+        /// - Parameter featureCollectionTable: Holds the tapped location feature
+        private func performGeoprocessing(_ featureCollectionTable: FeatureCollectionTable) async throws {
+            let params = GeoprocessingParameters(executionType: .synchronousExecute)
+            // Sets the parameters spatial reference to the point tapped on the map.
+            params.processSpatialReference = featureCollectionTable.spatialReference
+            params.outputSpatialReference = featureCollectionTable.spatialReference
+            // Create a feature with the feature collection table which has the reference to the tapped location.
+            let geoprocessingFeature = GeoprocessingFeatures(features: featureCollectionTable)
+            // Sets the observation point to the tapped location.
+            params.setInputValue(geoprocessingFeature, forKey: "Input_Observation_Point")
+            // Creates geoprocessing job and kicks off process of getting viewshed for the point.
+            geoprocessingJob = geoprocessingTask.makeJob(parameters: params)
+            geoprocessingJob?.start()
+            // Get the result of the geoprocessing job asynchronously.
+            let result = await geoprocessingJob?.result
+            defer { geoprocessingJob = nil }
+            switch result {
+            case .success(let output):
+                if let resultFeatures = output.outputs["Viewshed_Result"] as? GeoprocessingFeatures {
+                    processFeatures(resultFeatures: resultFeatures)
+                }
+            case .failure(let error):
+                throw error
+            case .none:
+                // This case should never execute.
+                break
+            }
```

Comment:
A few comment on this section

- Instead of using the `result`, use `output` of the `geoprocessingJob` can save you the switch-case
- Instead of unwrapping the feature set in the `processFeatures` method, you can unwrap it here, then pass the non-optional feature set into the following method. Therefore, `processFeatures` will be more intuitive.
- Instead of using the instance property `geoprocessingJob` to hold the job reference, I tend to use a local variable `job`. It saves several conditional chaining. You can also force unwrap if you prefer that style.

```swift
let job = geoprocessingTask.makeJob(parameters: params)
geoprocessingJob = job
defer { geoprocessingJob = nil }
job.start()
// Get the result of the geoprocessing job asynchronously.
let output = try await job.output
if let resultFeatures = output.outputs["Viewshed_Result"] as? GeoprocessingFeatures,
   let featureSet = resultFeatures.features {
    processFeatures(in: featureSet)  // private func processFeatures(in featureSet: FeatureSet)
}
```